### ISSUE TITLE
[Javascript] Web Ui Developers wanted

### ISSUE THREAD

**Issue by defnax** [RetroShare/RetroShare#2491]
Hi,
ouer Webui is not realy functional or yet useable.
We need some active javascript developers.

thanks to @Kumaravinash9 & @gunjan543 for helping us on the webui.

At the moment there are some todos

- Webui https://github.com/RetroShare/RSNewWebUI/issues/65
- Mail https://github.com/RetroShare/RSNewWebUI/issues/69
- Chat rooms https://github.com/RetroShare/RSNewWebUI/issues/42 
- Network View https://github.com/RetroShare/RSNewWebUI/issues/36 
- Files View https://github.com/RetroShare/RSNewWebUI/issues/44 F
- Private Chat (with Node), Distant Chat (with Contacts/People)
- we need a useable 1:1 chats too

- Show private Chat with Nodes in Network -> Chat Conversation/History
- Show distant Chat with Contacts/People People ->Conversations List -> Conversation/History
- Idea for Network Node Chat, Contacts/People Distant Chat and Chat Rooms, would be great we have the same design structure when possible.

Later:
* Wire/Wiki

**Repository:**
https://github.com/RetroShare/RSNewWebUI

**Readme (JSON API):**
https://github.com/RetroShare/libretroshare/blob/master/src/jsonapi/README.adoc

**RetroShare interface headers:**
https://github.com/RetroShare/libretroshare/tree/master/src/retroshare

**Comment by felisucoibi** [comment]
what is already done?

**Comment by zeners** [comment]
done: some chat, (some) reading mail, some file download

**Comment by csoler** [comment]
Most of the hard-core stuff (communication with retroshare) is done. What remains is mostly JS code to display the information in a nice layout.

**Comment by DeMoN100** [comment]
Please however the GUI is designed make it all changeable by preference of color instead of using skins. Use a color selector & allow the user to customize the gui how they like it.

**Comment by gunjan543** [comment]
Hello @defnax, I would like to contribute once again. Please let me know how I can start contributing to this.

**Comment by defnax** [comment]
look here on the issues:
https://github.com/RetroShare/RSNewWebUI/issues
maybe get connected with @zelfroster [@zelfroster](https://github.com/zelfroster)

the Board page is not yet finished, maybe you can implement the boards view?
https://github.com/RetroShare/RSNewWebUI/issues/65

**Comment by zelfroster** [comment]
> look here on the issues: https://github.com/RetroShare/RSNewWebUI/issues maybe get connected with @zelfroster [@zelfroster](https://github.com/zelfroster)
> 
> the Board page is not yet finished, maybe you can implement the boards view? [RetroShare/RSNewWebUI#65](https://github.com/RetroShare/RSNewWebUI/issues/65)

Sorry for the late reply, and you can ask for any help if needed @gunjan543 

As I have moved the CSS code to SCSS, you will need to run a few commands before starting development.

First, you need to install [pnpm](https://pnpm.io/installation) (package manager) in your machine.
Then, run `cd webui-src` from the root folder of your forked version of [RSNewWebUI](https://github.com/RetroShare/RSNewWebUI) in your local machine.
And then run `pnpm install` to install all the dependencies, which is just `sass`.

After this everything is done and whenever you start developing just run this command `pnpm watch` so that it would start compiling all the SCSS into CSS as you are working on it.

Finally, when your work is done, then before committing your changes, you must run `pnpm build` so that it would build a minified version of the CSS in the `webui-src/styles/app.css`.

Edit :- I will soon add this in the README as well 😅

**Comment by defnax** [comment]
@zelfroster maybe can you write her a email? was nice when you get little help from her :)

**Comment by zelfroster** [comment]
> @zelfroster maybe can you write her a email? was nice when you get little help from her :)

Ok, I will do it. Thanks for suggesting 😁

**Comment by Uplix** [comment]
is it too late to join and help out?

**Comment by defnax** [comment]
Hello, you are welcome we has no more webui dev

**Comment by csoler** [comment]
Indeed. You're very welcome to help.

**Comment by lunchfang** [comment]
👋 Hi there!

I'm interested in working on this task. Before I start, I'd like to confirm a few details:

1. **Bounty Amount**: Is the bounty still TBD?
2. **Acceptance Criteria**: What exactly needs to be delivered? Any specific requirements?
3. **Timeline**: Any deadline or preferred completion time?
4. **Testing**: Are there specific test cases or environments I should use?
5. **Payment**: What's the preferred payment method once completed?

Please confirm and I'll get started right away! 🚀

---
*Automated message from Atlas Bounty Hunter*

**Comment by defnax** [comment]
> is it too late to join and help out?
Here you can create pull requests

https://github.com/RetroShare/RSNewWebUI/pulls

**Comment by JsamsIvyTech** [comment]
I can possibly help and would love to contribute so I can learn more
